data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy

```diff
@@ -429,6 +433,46 @@ interface MyRepository {
             ]
     }
 
+    void "test build create index from table annotation"() {
+        when:
+        QueryBuilder encoder = new SqlQueryBuilder()
+        def statements = encoder.buildCreateTableStatements(getRuntimePersistentEntity(ShipmentWithIndex))
+
+        then:
+        statements[0] == 'CREATE TABLE "shipment_with_index" ("shipment_id" BIGINT PRIMARY KEY AUTO_INCREMENT,"field" VARCHAR(255) NOT NULL,"taxCode" VARCHAR(255) NOT NULL);'
+        statements[1] == 'CREATE UNIQUE INDEX idx_shipment_with_index_field_taxcode ON "shipment_with_index" (field, taxCode);'
+    }
+
+    void "test build create index from field annotation"() {
+        when:
+        QueryBuilder encoder = new SqlQueryBuilder()
+        def statements = encoder.buildCreateTableStatements(getRuntimePersistentEntity(ShipmentWithIndexOnFields))
+
+        then:
+        statements[0] == 'CREATE TABLE "shipment_with_index_on_fields" ("shipment_id" BIGINT PRIMARY KEY AUTO_INCREMENT,"field" VARCHAR(255) NOT NULL,"taxCode" VARCHAR(255) NOT NULL);'
+        statements[1] == 'CREATE UNIQUE INDEX idx_shipment_with_index_on_fields_field ON "shipment_with_index_on_fields" (field);CREATE INDEX idx_shipment_with_index_on_fields_taxcode ON "shipment_with_index_on_fields" (taxCode);'
+    }
+
+    void "test build create index from field annotation with composite indexes"() {
+        when:
+        QueryBuilder encoder = new SqlQueryBuilder()
+        def statements = encoder.buildCreateTableStatements(getRuntimePersistentEntity(ShipmentWithIndexOnFieldsCompositeIndexes))
+
+        then:
+        statements[0] == 'CREATE TABLE "shipment_with_index_on_fields_composite_indexes" ("shipment_id" BIGINT PRIMARY KEY AUTO_INCREMENT,"field" VARCHAR(255) NOT NULL,"taxCode" VARCHAR(255) NOT NULL);'
+        statements[1] == 'CREATE UNIQUE INDEX idx_shipment_with_index_on_fields_composite_indexes_field_taxcode ON "shipment_with_index_on_fields_composite_indexes" (field, taxCode);'
+    }
+
+    void "test build create index from index class annotation"() {
+        when:
+        QueryBuilder encoder = new SqlQueryBuilder()
+        def statements = encoder.buildCreateTableStatements(getRuntimePersistentEntity(ShipmentWithIndexOnClass))
+
+        then:
+        statements[0] == 'CREATE TABLE "shipment_with_index_on_class" ("shipment_id" BIGINT PRIMARY KEY AUTO_INCREMENT,"field" VARCHAR(255) NOT NULL,"taxCode" VARCHAR(255) NOT NULL);'
+        statements[1] == 'CREATE UNIQUE INDEX idx_shipment_with_index_on_class_field ON "shipment_with_index_on_class" (field);CREATE INDEX idx_shipment_with_index_on_class_taxcode ON "shipment_with_index_on_class" (taxCode);'
```

Comment:
I wonder if the syntax is the same for all the DBs we support.

Reply:
Good point I will check that, however that should be standard SQL-99.

Reply:
I had a look at:
https://dev.mysql.com/doc/refman/8.0/en/create-index.html
https://www.postgresql.org/docs/9.1/sql-createindex.html
https://www.oracletutorial.com/oracle-index/oracle-create-index/

And the basic syntax is the same for all. 
PostgreSQL supports also the keyword USING, MySQL has some advanced features like  _ON (name(10))_ or 
_ON (name + surname)_ but the foundation is the same for all the three of them.